plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/DelegatingRestHandler.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.shiro;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Delegating RestHandler that delegates all implementations to original handler
+ */
+public class DelegatingRestHandler implements RestHandler {
+
+    protected final RestHandler delegate;
+
+    public DelegatingRestHandler(RestHandler delegate) {
```

Comment:
What is the purpose of this? How do we benefit from having what is effectively a noop wrapper?

Reply:
You can find the usage here: https://github.com/opensearch-project/OpenSearch/pull/15430/files#diff-3743000bc93e986fe6a3d456425583c1646199f9c215b96477e0ccefd679e9b7R112-R117

The purpose of this is to ensure that the RestHandlerWrapper that the Shiro plugin creates properly delegates to the underlying original handler. 

Its relevant for this block of code:

```
@Override
public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
    return AuthcRestHandler::new;
}
```

where the AuthcRestHandler is the wrapper that overrides `handleRequest` to perform auth. If auth succeeds, it will call the original handler's `handleRequest, otherwise it responds with UNAUTHORIZED to the caller.

Instead of iterating through all default methods of the BaseHandler to ensure that AuthcRestHandler properly delegates, it subclasses DelegatingRestHandler to ensure that all of the original handler's methods are called.

Reply:
Similar concerns as @stephen-crawford : why do we need this class? how it is different from `RestHandler::Wrapper` class [1]?

[1] https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/RestHandler.java#L137

Reply:
I'll switch this to use RestHandler::Wrapper. 

This is a class that was added to the security plugin which creates a similar RestHandler Wrapper and you can find the reason for this class in the issue description here: https://github.com/opensearch-project/security/issues/3422

Reply:
Removed DelegatingRestHandler in favor of using RestHandler.Wrapper